src/main/java/org/jenkinsci/plugins/artifactdeployer/MatrixArtifactDeployerPublisher.java

```diff
@@ -0,0 +1,60 @@
+package org.jenkinsci.plugins.artifactdeployer;
+
+import hudson.Extension;
+import hudson.Launcher;
+import hudson.matrix.MatrixAggregatable;
+import hudson.matrix.MatrixAggregator;
+import hudson.matrix.MatrixBuild;
+import hudson.matrix.MatrixConfiguration;
+import hudson.matrix.MatrixProject;
+import hudson.matrix.MatrixRun;
+import hudson.model.AbstractBuild;
+import hudson.model.AbstractProject;
+import hudson.model.BuildListener;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Extends ArtifactDeployerPublisher to support Matrix projects and implement MatrixAggregatable.
+ * @author <a href="mailto:nicolas.deloof@gmail.com">Nicolas De Loof</a>
+ */
+public class MatrixArtifactDeployerPublisher extends ArtifactDeployerPublisher implements MatrixAggregatable {
+
+    @DataBoundConstructor
+    public MatrixArtifactDeployerPublisher(List<ArtifactDeployerEntry> deployedArtifact, boolean deployEvenBuildFail) {
+        super(deployedArtifact, deployEvenBuildFail);
+    }
```

Comment:
And this should also preferably have a `writeReplace` so that its serial form remains that of `ArtifactDeployerPublisher`, i.e., so that `MatrixArtifactDeployerPublisher` only ever exists in memory.